### PR TITLE
Avoid using same discovery port (default 5353) to send queries and listen to responses

### PR DESCRIPTION
**Pull request by diasdavid**
Hi @mafintosh !

Found a bug when making go mdns and node multicast-dns interop. tl;dr; Since node was always using the same port for both listening and querying (default: 5353), when the go impl tried to send the response to the query, back to node and in the same machine, the node implementation never received the message because it would be a message srcPort === dstPort and with same IP.

This patch essentially only makes that queries are sent from a random port, while responses to those queries continue being made from 5353 (or opts.port).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,57 +12,63 @@ module.exports = function (opts) {
...
-  bind()
+    return bind
+  }
+
+  binder(port)()
+   // to avoid sending on the same port as we listen for queries
 
   that.send = function (packet, cb) {
-    bind(function (err, socket) {
+    binder(0)(function (err, socket) {
```
Comment (mafintosh) [inline index.js]: this pushes a new socket to the sockets array every time we send a message

Comment (diasdavid) [inline index.js]: good point. Fixed in:
https://github.com/diasdavid/multicast-dns/commit/583fca35691e943ada4149b7a453a5c82c6939d8

**Comment by mafintosh** [comment]
Thanks! I cleaned up the impl a bit (see 684071e) and released this as 3.0.0

**Comment by diasdavid** [comment]
awesome, thank you @mafintosh !

**Comment by mafintosh** [comment]
@diasdavid btw is the go impl using multicast_loopback? (https://github.com/mafintosh/multicast-dns/blob/master/index.js#L51)

**Comment by bitmonki** [comment]
I wonder if the issue mentioned above about not receiving the message isn't a bug elsewhere, perhaps in the 'go impl'?

I mention this because RFC 6762 ('Multicast DNS'), Section 5.2, page 10 states:

>   A compliant Multicast DNS querier, which implements the rules
   specified in this document, MUST send its Multicast DNS queries from
   UDP source port 5353 (the well-known port assigned to mDNS), and MUST
   listen for Multicast DNS replies sent to UDP destination port 5353 at
   the mDNS link-local multicast address (224.0.0.251 and/or its IPv6
   equivalent FF02::FB).

Surely this would not be the case if it resulted in the issue described above?

**Comment by bitmonki** [comment]
P.S. *GREAT* job on this, btw!!! Thanks sooo much for multicast-dns, I am using it to write an actual DNSSD browser on JXCore.

**Pull request merged**